app/models/commit.rb

```diff
@@ -83,6 +83,18 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    commit_was_already_successful = success?
+    new_status = statuses.create!(status_attributes)
+    reload
+    if (!commit_was_already_successful && success?) ||
+       (!new_status.success? && !new_status.pending? &&
+        stack.filter_visible_statuses([new_status]).any? && stack.filter_meaningful_statuses([new_status]).any?)
```

Comment:
Yes, using `significant_status` made it simpler - but not quite as simple as your suggestion. We need to make sure that we don't fire the old failure status if an unimportant failure status comes in (which wouldn't cause `significant_status` to actually change).